web-api/workflow-terraform/migration-cron/main/lambdas/migration-status.js

```diff
@@ -0,0 +1,102 @@
+const {
+  approvePendingJob,
+  cancelWorkflow,
+} = require('../../../../../shared/admin-tools/circleci/circleci-helper');
+const {
+  getMetricStatistics,
+  getMigrationQueueIsEmptyFlag,
+  getSqsQueueCount,
+  putMigrationQueueIsEmptyFlag,
+} = require('../../../../../shared/admin-tools/aws/migrationWaitHelper');
+
+const apiToken = process.env.CIRCLE_MACHINE_USER_TOKEN;
+const workflowId = process.env.CIRCLE_WORKFLOW_ID;
+const dlQueueUrl = `https://sqs.us-east-1.amazonaws.com/${process.env.AWS_ACCOUNT_ID}/migration_segments_dl_queue_${process.env.STAGE}`;
+const workQueueUrl = `https://sqs.us-east-1.amazonaws.com/${process.env.AWS_ACCOUNT_ID}/migration_segments_queue_${process.env.STAGE}`;
+const jobName = 'wait-for-migration';
+
+exports.handler = async (input, context) => {
+  const results = { migrateFlag: process.env.MIGRATE_FLAG };
+  if (results.migrateFlag !== 'true') {
+    await approvePendingJob({ apiToken, jobName, workflowId });
+    return context.succeed(results);
+  }
+
+  results.errorRate = await getSegmentErrorRate();
+  if (results.errorRate === -1) {
+    return context.succeed(results);
+  }
+  if (results.errorRate > 50) {
+    await cancelWorkflow({ apiToken, workflowId });
+    return context.succeed(results);
+  }
+
+  results.dlQueueCount = await getSqsQueueCount(dlQueueUrl);
+  if (results.dlQueueCount === -1) {
+    return context.succeed(results);
+  }
+  if (results.dlQueueCount > 0) {
+    await cancelWorkflow({ apiToken, workflowId });
+    return context.succeed(results);
+  }
+
+  // it's possible the queue has caught up but there are still more segments to process
+  // don't approve the job if this is the first consecutive time the queue is empty
+  results.totalActiveJobs = await getSqsQueueCount(workQueueUrl);
+  if (results.totalActiveJobs === -1) {
+    return context.succeed(results);
+  }
+  if (results.totalActiveJobs > 0) {
+    await putMigrationQueueIsEmptyFlag(false);
+  } else {
```

Comment:
this may need to `return context.succeed(results);`, and then you can remove the `else { ... }` block

Reply:
updated!